File: ios/TikTokWeb/Entrance.swift
import Foundation

/// 个人主页入口条上的一个入口：图标 + 名称 + 目标地址 + 路线标识。
///
/// 与安卓版 `Entrance.java` 一一对应：入口的增减、改名、换地址都在 `all()` 里一处完成，
/// 界面代码不需要改动，两端也不会出现「某个入口只在一边存在」的不一致。
struct Entrance {

    /// 图标造型（由 `EntranceIcon.image(size:)` 在运行时画出来）
    let icon: EntranceIcon
    /// 名称文案，需与安卓 `app/res/values/strings.xml` 里的 entry_* 保持一致
    let title: String
    /// 目标地址（已含 route 参数，用户信息由 MallViewController 再追加）
    let url: String
    /// 路线标识，随地址与用户信息一起传给网页
    let route: String

    /// 参考包在个人主页放四个入口，这里保持同样的数量和顺序
    static func all() -> [Entrance] {
        return [
            Entrance(icon: .shopCenter, title: "店铺中心",
                     url: Endpoints.shopCenter(), route: Endpoints.routeShopCenter),
            Entrance(icon: .showcase, title: "商品橱窗",
                     url: Endpoints.showcase(), route: Endpoints.routeShowcase),
            Entrance(icon: .orders, title: "我的订单",
                     url: Endpoints.orders(), route: Endpoints.routeOrders),
            Entrance(icon: .merchant, title: "商家入驻",
                     url: Endpoints.merchant(), route: Endpoints.routeMerchant),
        ]
    }
}

File: app/java/com/mallcenter/app/Entrance.java
package com.mallcenter.app;

/**
 * 个人主页入口条上的一个入口：图标 + 名称 + 目标地址 + 路线标识。
 *
 * 入口的增减、改名、换地址都在 {@link #all()} 里一处完成，界面代码不需要改动。
 */
final class Entrance {

    /** 图标资源（普通/按下两种状态已由 drawable 的 selector 处理） */
    final int iconRes;
    /** 名称文案资源 */
    final int labelRes;
    /** 目标地址（已含 route 参数，用户信息由 MallActivity 再追加） */
    final String url;
    /** 路线标识，随地址与用户信息一起传给网页 */
    final String route;

    Entrance(int iconRes, int labelRes, String url, String route) {
        this.iconRes = iconRes;
        this.labelRes = labelRes;
        this.url = url;
        this.route = route;
    }

    /**
     * 个人主页上的三个入口，顺序与参考包一致：店铺中心 / 商品橱窗 / 订单详情。
     *
     * 这里只放「用户最常点」的三项；商家入驻等偏B端的入口不在此列（地址仍保留在
     * {@link Endpoints#merchant()} 中，需要时可再往数组里追加，界面会自动按数量等分排布）。
     */
    static Entrance[] all() {
        return new Entrance[]{
                new Entrance(R.drawable.ic_entry_shop_center, R.string.entry_shop_center,
                        Endpoints.shopCenter(), Endpoints.ROUTE_SHOP_CENTER),
                new Entrance(R.drawable.ic_entry_showcase, R.string.entry_showcase,
                        Endpoints.showcase(), Endpoints.ROUTE_SHOWCASE),
                new Entrance(R.drawable.ic_entry_orders, R.string.entry_orders,
                        Endpoints.orders(), Endpoints.ROUTE_ORDERS),
        };
    }
}
